sdks/universal-router-sdk/src/entities/actions/uniswap.ts

```diff
@@ -62,8 +62,22 @@ export class UniswapTrade implements Command {
     else this.payerIsUser = true
   }
 
+  get isAllV4(): boolean {
+    let result = true
+    for (const swap of this.trade.swaps) {
+      result = result && swap.route.protocol == Protocol.V4
+    }
+    return result
+  }
+
   get inputRequiresWrap(): boolean {
-    return this.trade.inputAmount.currency.isNative
+    if (!this.isAllV4) {
+      return this.trade.inputAmount.currency.isNative
```

Comment:
you might want a second conditional here:

`this.trade.inputAmount.currency.isNative && this.trade.route.adjustedInput.isNative`
you'd need to double check that second statement, but this way we can provide ETH to WETH pools 🤷 ??